.github/workflows/review-check.yml

```diff
@@ -0,0 +1,56 @@
+name: Enforce Review Rules
+
+on:
+  pull_request:
+    types: [opened, edited, synchronize]
+
+jobs:
+  enforce-review-rules:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Get list of files changed
+      id: files
+      run: |
+        echo "::set-output name=files::$(git diff --name-only ${{ github.event.before }} ${{ github.sha }} | tr '\n' ' ')"
+    
+    - name: Check if PR modifies specific path
+      id: check-path
+      run: |
+        if echo "${{ steps.files.outputs.files }}" | grep -q '^src/'; then
+          echo "::set-output name=specific_path::true"
+        else
+          echo "::set-output name=specific_path::false"
+        fi
+
+    - name: Ensure required reviewers
+      id: ensure-reviewers
+      run: |
+        GITHUB_TOKEN=${{ secrets.GITHUB_TOKEN }}
+        PR_NUMBER=${{ github.event.pull_request.number }}
+        REPO=${{ github.repository }}
+        
+        # Fetch reviewers of the PR
+        REVIEWERS=$(curl -s -H "Authorization: token $GITHUB_TOKEN" \
+                    "https://api.github.com/repos/$REPO/pulls/$PR_NUMBER/reviews" \
+                    | jq -r '.[].user.login')
+
+        NUM_REVIEWERS=$(echo "$REVIEWERS" | wc -w)
```

Comment:
Unfortunately this just tells you the number of reviews, not the number of reviewers. See a double review by the same user here for example:
https://api.github.com/repos/cowdao-grants/cow-fee/pulls/21/reviews
I also suspect it treats comments and change requests as full reviews for this count.

Reply:
The suspicion is confirmed with this review.

<details><summary>api output</summary>

https://api.github.com/repos/cowprotocol/contracts/pulls/91/reviews

```json
[
  {
    "id": 2090907877,
    "node_id": "PRR_kwDOHF14wc58oLjl",
    "user": {
      "login": "fedgiac",
      "id": 58218759,
      "node_id": "MDQ6VXNlcjU4MjE4NzU5",
      "avatar_url": "https://avatars.githubusercontent.com/u/58218759?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/fedgiac",
      "html_url": "https://github.com/fedgiac",
      "followers_url": "https://api.github.com/users/fedgiac/followers",
      "following_url": "https://api.github.com/users/fedgiac/following{/other_user}",
      "gists_url": "https://api.github.com/users/fedgiac/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/fedgiac/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/fedgiac/subscriptions",
      "organizations_url": "https://api.github.com/users/fedgiac/orgs",
      "repos_url": "https://api.github.com/users/fedgiac/repos",
      "events_url": "https://api.github.com/users/fedgiac/events{/privacy}",
      "received_events_url": "https://api.github.com/users/fedgiac/received_events",
      "type": "User",
      "site_admin": false
    },
    "body": "Some content on the new workflow, the rest looks good.",
    "state": "COMMENTED",
    "html_url": "https://github.com/cowprotocol/contracts/pull/91#pullrequestreview-2090907877",
    "pull_request_url": "https://api.github.com/repos/cowprotocol/contracts/pulls/91",
    "author_association": "CONTRIBUTOR",
    "_links": {
      "html": {
        "href": "https://github.com/cowprotocol/contracts/pull/91#pullrequestreview-2090907877"
      },
      "pull_request": {
        "href": "https://api.github.com/repos/cowprotocol/contracts/pulls/91"
      }
    },
    "submitted_at": "2024-05-31T14:38:07Z",
    "commit_id": "fe4be933d3769a34cbc35cf386ac9dc70a0beb93"
  }
]
```
</details>